### ISSUE TITLE
Please add automatic detection of caffeine, coffee; alcohol in the product.

### ISSUE THREAD

**Issue by Slawek234** [openfoodfacts/openfoodfacts-server#2184]
Please add automatic detection of caffeine, coffee in the product. Examples:
- https://world.openfoodfacts.org/product/3760039982018/creme-glacee-cafe-pur-arabica-angelys-l-angelys
- https://de.openfoodfacts.org/produkt/5449000000996/cola-erfrischungsgetrank-coca-cola
- https://pl.openfoodfacts.org/product/5902891280231/cappuccino-z-magnezem-mokate
- https://world.openfoodfacts.org/product/8996001320013/kopiko-original-mayora

**Comment by aleene** [comment]
Why? what do you look for?

**Comment by Slawek234** [comment]
Because many people are sensitive to caffeine: pregnant women, people with hypertension problems.
In my opinion, there is no additional information that the product contains caffeine.

**Comment by zigouras** [comment]
I agree tat this would be a very useful feature.  Personally, I am often looking to see if there is caffeine in the product.  This is very important.

**Comment by VaiTon** [comment]
Where should we insert this kind of warning?

**Comment by stephanegigandet** [comment]
I would like to understand the need and possible implementation a bit better. Would we just flag products that have "coffee" (with the exception of decaffeinated coffee) or "caffeine" in the ingredients? Is there more to it? Are there harmless looking products that contain caffeine? (that are not products that obviously contain coffee)

**Comment by VaiTon** [comment]
> I would like to understand the need and possible implementation a bit better. Would we just flag products that have "coffee" (with the exception of decaffeinated coffee) or "caffeine" in the ingredients? Is there more to it? Are there harmless looking products that contain caffeine? (that are not products that obviously contain coffee)

We should flag every product that has a lot of caffeine or that is coffee. 
The warning should be something like what we do for alcohol but a little bit less concerning, like:

"This product contains a lot of caffeine. Consume with moderation."
